Jint/Native/String/StringPrototype.cs

```diff
@@ -793,5 +794,45 @@ private JsValue Pad(JsValue thisObj, JsValue[] arguments, bool padStart)
 
             return padStart ? $"{padString.Substring(0, targetLength)}{s}" : $"{s}{padString.Substring(0, targetLength)}";
         }
+
+        /// <summary>
+        /// https://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.startswith
+        /// </summary>
+        /// <param name="thisObj"></param>
+        /// <param name="arguments"></param>
+        /// <returns></returns>
+        private JsValue StartsWith(JsValue thisObj, JsValue[] arguments)
+        {
+            TypeConverter.CheckObjectCoercible(Engine, thisObj);
+
+            var s = TypeConverter.ToString(thisObj);
+
+            var searchString = arguments.At(0);
+            if (searchString == Null.Instance)
+            {
+                searchString = Null.Text;
+            }
+            else
+            {
+                if (searchString.IsRegExp())
+                {
+                    throw new JavaScriptException(Engine.TypeError);
+                }
+            }
+            
+            var searchStr = TypeConverter.ToString(searchString);
+
+            var pos = TypeConverter.ToInt32(arguments.At(1));
+
+            var len = s.Length;
+            var start = System.Math.Min(System.Math.Max(pos, 0), len);
+            var searchLength = searchStr.Length;
+            if (searchLength + start > len)
+            {
+                return false;
+            }
+
+            return s.Remove(0, start).StartsWith(searchStr);
```

Comment:
Better to use IndexOf(start, str) == 0

Reply:
Cool. Thanks for the feedback. Will update asap :-)